examples/direct_fidelity_estimation.ipynb

```diff
@@ -0,0 +1,228 @@
+    "\n",
+    "The code is optimized for Clifford circuits using tableaux\n",
+    "* Improved Simulation of Stabilizer Circuits (https://arxiv.org/pdf/quant-ph/0406196.pdf)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "0: ───@───Z───────\n",
+      "      │\n",
+      "1: ───┼───────X───\n",
+      "      │       │\n",
+      "2: ───X───H───@───\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Import Cirq, DFE, and create a circuit\n",
+    "\n",
+    "import cirq\n",
+    "import direct_fidelity_estimation as dfe\n",
+    "\n",
+    "qubits = cirq.LineQubit.range(3)\n",
+    "circuit = cirq.Circuit(cirq.CNOT(qubits[0], qubits[2]),\n",
+    "                       cirq.Z(qubits[0]),\n",
+    "                       cirq.H(qubits[2]),\n",
+    "                       cirq.CNOT(qubits[2], qubits[1]))\n",
+    "\n",
+    "print(circuit)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We then create a sampler. For this example, we use a simulator but the code can accept a hardware sampler.\n",
+    "noise = cirq.ConstantQubitNoiseModel(cirq.depolarize(0.1))\n",
+    "sampler = cirq.DensityMatrixSimulator(noise=noise)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Estimated fidelity: 0.53\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We run the DFE:\n",
+    "estimated_fidelity, intermediate_results = dfe.direct_fidelity_estimation(\n",
+    "        circuit,\n",
+    "        qubits,\n",
+    "        sampler,\n",
+    "        n_measured_operators=None,\n",
+    "        samples_per_term=0)\n",
+    "\n",
+    "print('Estimated fidelity: %.2f' % (estimated_fidelity))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# What is happening under the hood?\n",
+    "Now, let's look at the `intermediate_results` and correlate what is happening in the code with the papers. The definition of fidelity is:\n",
+    "$$\n",
+    "F = F(\\hat{\\rho},\\hat{\\sigma}) = tr \\left(\\hat{\\rho} \\hat{\\sigma}\\right)\n",
+    "$$\n",
+    "where $\\hat{\\rho}$ is the theoretical pure state and $\\hat{\\sigma}$ is the actual state. The idea of DFE is to write fidelity as:\n",
+    "$$F= \\sum _i \\frac{\\rho _i \\sigma _i}{d}$$\n",
+    "\n",
+    "where $d=4^{\\mathit{number-of-qubits}}$, $\\rho _i = tr \\left( \\hat{\\rho} P_i \\right)$, and $\\sigma _i = tr \\left(\\hat{\\sigma} P_i \\right)$. Each of the $P_i$ is a Pauli operator. We can then finally rewrite the fidelity as:\n",
+    "\n",
+    "$$F= \\sum _i Pr(i) . \\frac{\\sigma _i}{\\rho_i}$$\n",
```

Comment:
I don't think `.` is what you want here? `\cdot`? or just nothing and have the multiplication be implied

Reply:
Done

Reply:
Done